custom_components/tuya_local/devices/himox_h06_purifier.yaml

```diff
@@ -0,0 +1,97 @@
+name: himox h06
+products:
+ - id: 80256465f4cfa2c5ec84
+primary_entity:
+  entity: fan
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+    - id: 4
+      type: string
+      name: speed
+      mapping:
+        - dps_val: "low"
+          value: 33.34
+        - dps_val: "mid"
+          value: 66.67
+        - dps_val: "high"
+          value: 100
+secondary_entities:
+  - entity: sensor
+    name: active_filter_life
+#   icon: mdi:filter
+    dps:
+      - id: 5
+        type: integer
+        name: sensor
+        readonly: true
+        unit: "%"
+        class: measurement
+        range:
+          min: 0
+          max: 100
+        scale: 0
+        step: 1
+  - entity: light
+    name: aq_indicator
+    dps:
+      - id: 8
+        type: boolean
+        name: switch
+  - entity: switch
+    name: filter_reset
+    dps:
+      - id: 11
+        type: boolean
+        name: switch
+  - entity: select
+    name: timer
+    icon: "mdi:timer"
+    category: config
+    dps:
+      - id: 18
+        name: option
+        type: string
+        mapping:
+          - dps_val: cancel
+            value: "Off"
+          - dps_val: 4h
+            value: "4 hours"
+          - dps_val: 8h
+            value: "8 hours"
+  - entity: sensor
+    name: countdown_left
+    dps:
+      - id: 19
+        type: integer
+        name: sensor
+        readonly: true
+        unit: "min"
+        class: measurement
+        range:
+          min: 0
+          max: 480
+        scale: 0
+        step: 1
+  - entity: sensor
+    name: aq_indicator
+    dps:
+      - id: 22
+        type: string
+        name: sensor
+        readonly: true
+  - entity: select
```

Comment:
I think the fan entity's preset_mode is still available, which might be better for this.